ibc-core/ics02-client/types/src/events.rs

```diff
@@ -56,31 +56,31 @@ impl TryFrom<abci::EventAttribute> for ClientIdAttribute {
     type Error = ClientError;
 
     fn try_from(value: abci::EventAttribute) -> Result<Self, Self::Error> {
+        if let Ok(key_str) = value.key_str() {
+            if key_str != CLIENT_ID_ATTRIBUTE_KEY {
+                return Err(ClientError::InvalidAttributeKey {
+                    attribute_key: key_str.to_string(),
+                });
+            }
+        } else {
+            return Err(ClientError::InvalidAttributeKey {
+                attribute_key: String::new(),
+            });
+        }
+
         value
-            .key_str()
-            .map(|key| {
-                if key != CLIENT_ID_ATTRIBUTE_KEY {
-                    return Err(ClientError::InvalidAttributeKey {
-                        attribute_key: key.to_string(),
-                    });
-                }
-                value
-                    .value_str()
-                    .map(|value| {
-                        Ok(ClientIdAttribute {
-                            client_id: ClientId::from_str(value).map_err(|_| {
-                                ClientError::InvalidAttributeValue {
-                                    attribute_value: value.to_string(),
-                                }
-                            })?,
-                        })
-                    })
-                    .map_err(|_| ClientError::InvalidAttributeValue {
-                        attribute_value: String::new(),
-                    })?
+            .value_str()
+            .map(|value| {
+                Ok(ClientIdAttribute {
+                    client_id: ClientId::from_str(value).map_err(|_| {
+                        ClientError::InvalidAttributeValue {
+                            attribute_value: value.to_string(),
+                        }
+                    })?,
+                })
```

Comment:
I'd prefer to define a new variable for the attribute field values:
```suggestion
                let client_id = ClientId::from_str(value).map_err(|_| {
                    ClientError::InvalidAttributeValue {
                        attribute_value: value.to_string(),
                    }
                })?;
                
                Ok(ClientIdAttribute { client_id })
```